runtime/doc/channels.txt

```diff
@@ -0,0 +1,89 @@
+1. Introduction						    *channels-intro*
+
+Channels are nvim's way of communicating with external processes.
+
+There are several ways to open a channel:
+
+  1. Through stdin/stdout when `nvim` is started with `--headless`, and a startup
+     script or --cmd  commad opens the stdio channel using |stdioopen()|.
+
+  2. Through stdin, stdout and stderr of a process spawned by |jobstart()|.
+
+  3. Through the PTY master end of a PTY opened with
+     `jobstart(..., {'pty': v:true})` or |termopen()|.
+
+  4. By connecting to a TCP/IP socket or named pipe with |sockconnect()|.
+
+  5. By another process connecting to a socket listened by nvim. This only
+     supports RPC channels, see |rpc-connecting|.
+
+Channels can operate in two different modes, a mode using the |rpc| protocol,
+based on the msgpack-rpc standard, and in "bytes" mode, where scripts can read
+and write raw bytes over the channel. As a special case, even if a job channel
+uses rpc, bytes can still be read over its stderr. Similarily, only bytes can
+be written to nvim's own stderr.
+
+==============================================================================
+2. Reading and writing raw bytes			      *channels-bytes*
+
+By default, channels opened by vimscript functions will operate with raw
+bytes. A callback function will be invoked with data read from the channel.
+By default, the callback will invoked immediately when data is available, to
+facilitate interactive communication. The same callback will then be invoked
+with empty data, to indicate that the stream reached EOF.  Alternatively an
+option can be set to invoke the callback only when the underlying stream
+reaches EOF, and will then be passed in complete output.  This is helpful when
+only the complete output is useful, and not partial data.
+
+Data can be sent to the channel using the |chansend()| function.
+
+
+TODO: simple example using cat or something
+
+TODO: another example using stdout_buffered
+
+As a special case is PTY channels opened by  `jobstart(..., {'pty': v:true})` .
+No preprocessing of ANSI escape sequences is done, these will be sent raw to the callback. However,
+change of PTY size can be signaled to the slave using |jobresize()|. See also
+|terminal-emulator|.
+
+==============================================================================
+3. Communicating using msgpack-rpc			      *channels-rpc*
+
+TODO: figure out good split with msgpack_rpc.txt
+
+==============================================================================
+4. Using the stdio channel			*channels-stdio* *--headless*
+
+When invoked normally, nvim will use stdin and stdout to interact with the
+user over the terminal interface (TUI). However when invoked with
+`--headless`, the TUI is not started and stdin and stdout can be used as a
+channel. To open the stdio channel |stdioopen()| must be called during
+|startup|, as there later will be no way of invoking a command. Here is a
```

Comment:
With `` `startup` `` (instead of `|startup|`) we can jump to the `startup` help.